src/tls13.c

```diff
@@ -1681,9 +1681,13 @@ static int EncryptTls13(WOLFSSL* ssl, byte* output, const byte* input,
                 #endif
 
                     nonceSz = AESGCM_NONCE_SZ;
-                    ret = wc_AesGcmEncrypt(ssl->encrypt.aes, output, input,
-                        dataSz, ssl->encrypt.nonce, nonceSz,
-                        output + dataSz, macSz, aad, aadSz);
+                    ret = wc_AesGcmSetExtIV(ssl->encrypt.aes,
+                            ssl->encrypt.nonce, nonceSz);
+                    if (ret == 0) {
+                        ret = wc_AesGcmEncrypt_ex(ssl->encrypt.aes, output,
```

Comment:
Why isn't the TLS 1.2 case this simple?

Reply:
For TLSv1.2, we set the IV once, when the keys are generated. The other changes were to keep the old FIPS working with the code. For TLSv1.3, we have to set the IV on every encrypt because according to the TLSv1.3 spec, the sequence number is XORed into the nonce, and then it is given the encrypt. NIST doesn't currently have guidance reconciling how TLSv1.3 handles its IV versus how AES-GCM handles the IV.